openedx/core/djangoapps/util/row_delete.py

```diff
@@ -0,0 +1,101 @@
+"""
+Code to delete rows from a table within a Django mgmt command using best practices.
+"""
+
+import logging
+import time
+from django.core.management.base import CommandError, BaseCommand
+from django.db import transaction
+
+log = logging.getLogger(__name__)
+
+
+def delete_rows(
+        model_mgr,
+        table_name,
+        chunk_size,
+        sleep_between
+    ):
+    """
+    Deletes *ALL* rows from table, chunking the deletes to avoid long table/row locks.
+
+    Args:
+        model_mgr (django.db.models.manager.Manager): Django ORM mgr for the table's model.
+        table_name (str): Name of table from which to delete all rows.
+        chunk_size (int): Number of rows to delete in each transaction.
+        sleep_between (float): Number of seconds to sleep between transactions.
+    """
+    if chunk_size <= 0:
+        raise CommandError('Only positive chunk size is allowed ({}).'.format(chunk_size))
+    if sleep_between < 0:
+        raise CommandError('Only non-negative sleep between seconds is allowed ({}).'.format(sleep_between))
+
+    # The "as id" below fools Django raw query into thinking the primary key is being queried.
+    min_max_ids = model_mgr.raw(
+        'SELECT MIN(id) as id, MAX(id) as max_id FROM {}'.format(table_name)
+    )[0]
+    min_id = min_max_ids.id
+    max_id = min_max_ids.max_id
+    if not min_id or not max_id:
+        log.info("No data exists in table %s - skipping.", table_name)
+        return
+    log.info("STARTED: Deleting around %s rows with chunk size of %s and %s seconds between chunk.",
+        max_id - min_id, chunk_size, sleep_between
```

Comment:
I know the message says "around %s rows" instead of giving an exact number, but adding one to the first argument would be slightly more accurate.

Reply:
I'll add one as suggested. But I'll keep the `around` in the log message as ID ranges might exist with missing records for various reasons.